docs/api.yaml

```diff
@@ -0,0 +1,641 @@
+          $ref: "#/components/responses/404"
+        429:
+          $ref: "#/components/responses/429"
+        500:
+          $ref: "#/components/responses/500"
+
+  /courses/teacher:
+    post:
+      tags:
+        - "Course"
+      summary: "Attribute a course to a teacher"
+      description: "Create a new relation between a teacher and a teaching unit"
+      operationId: "attributeTeacherCourse"
+      requestBody:
+        description: "The teacher and the teaching unit to attribute him"
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/TaughtCourse'
+      responses:
+        200:
+          description: "The created TaughtCourse"
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/TaughtCourse'
+        400:
+          $ref: "#/components/responses/400"
+        403:
+          $ref: "#/components/responses/403"
+        404:
+          $ref: "#/components/responses/404"
+        429:
+          $ref: "#/components/responses/429"
+        500:
+          $ref: "#/components/responses/500"
+
+  /activation:
+    patch:
+      tags:
+        - "Student"
+      summary: "Change activation status of a student"
+      description: ""
+      operationId: "changeActivation"
+      requestBody:
+        description: "The activation status of a student"
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Student'
+      responses:
+        200:
+          description: "The updated Student"
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Student'
+        400:
+          $ref: "#/components/responses/400"
+        403:
+          $ref: "#/components/responses/403"
+        404:
+          $ref: "#/components/responses/404"
+        429:
+          $ref: "#/components/responses/429"
+        500:
+          $ref: "#/components/responses/500"
+
+  /impersionation:
+    patch:
+      tags:
+        - "Student"
+      summary: "Allow a student to make changes or not"
+      description: ""
+      operationId: "allowChanges"
+      requestBody:
+        description: "The mode allowed to the student"
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Student'
+      responses:
+        200:
+          description: "The updated Student"
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Student'
+        400:
+          $ref: "#/components/responses/400"
+        403:
+          $ref: "#/components/responses/403"
+        404:
+          $ref: "#/components/responses/404"
+        429:
+          $ref: "#/components/responses/429"
+        500:
+          $ref: "#/components/responses/500"
+
```

Comment:
Again. These endpoints are too complicated. You totally forgot the `/students` and `/teachers` endpoints even though they are the central resources of the API. Please specify the following calls in priority and see what we can do with them before thinking about any additional endpoint:

- GET /students (for search with filter parameters, eg, by id, by date, by anything)
- POST /students (for batch creation)
- PATCH /students (for batch update)
- GET /teachers
- POST /teachers
- PATCH /teachers

Reply:
The endpoints are now changed to 4 endpoints : students,courses,teachers and grades.
I noticed that resources of an object with a relation ONE TO ONE/MANY can be easily added with PATCH verb but when I try with an object with a relation MANY TO MANY,  it's not the same so that's why I created the endpoint "grades", using a POST verb.